File: Coding/Blend/算法题/LeetCode/121_买卖股票的最佳时机.java
/**
 * 执行用时：252ms
 * 内存消耗：38.1mb
 * 
 */
class Solution {
    public int maxProfit(int[] prices) {
        int maxPrice = 0;
        for (int i = 0; i < prices.length - 1; i++)
        {
            for (int j = i + 1; j < prices.length; j++)
            {
                maxPrice = Math.max(maxPrice, prices[j] - prices[i]);
            }
        }
        return maxPrice;
    }
}

/**
 * 方法二：
 */

File: Coding/normal/python/projects/DemoProjects/PythonDemos1/com/smx/pythondemos1/algorithm/leetcodes/121_买卖股票的最佳时机.py
class Solution:
    def maxProfit(self, prices) -> int:
        if not prices:
            return 0
        max_profit = 0
        in_price = prices[0]
        for index in range(1, len(prices)):
            if prices[index] < in_price:
                in_price = prices[index]
            elif prices[index] > in_price:
                max_profit = max(max_profit, prices[index] - in_price)
        return max_profit


s = Solution()
ret = s.maxProfit([7, 6, 4, 3, 1])
print(ret)
